src/main/java/org/fogbowcloud/manager/core/datastructures/OrderStateTransitioner.java

```diff
@@ -0,0 +1,73 @@
+package org.fogbowcloud.manager.core.datastructures;
+
+import org.apache.log4j.Logger;
+import org.fogbowcloud.manager.core.models.linkedList.SynchronizedDoublyLinkedList;
+import org.fogbowcloud.manager.core.models.orders.Order;
+import org.fogbowcloud.manager.core.models.orders.OrderState;
+
+public class OrderStateTransitioner {
+
+    private static final Logger LOGGER = Logger.getLogger(OrderStateTransitioner.class);
+
+    /**
+     * Updates the state of the given order. This method is not
+     * thread-safe. The caller must assure that only one
+     * thread can call it at the same time.
```

Comment:
Complement this comment with the following:
 * This is done by creating a synchronized block on the order object that
 * is passed as parameter to updateState(). The reason the synchronized
 * block is created in the code calling this method is because, usually the
 * order will be inspected before the call to updateState() is made, and
 * this needs to be done within a synchronized block, since several threads
 * may attempt to process the same order at the same time.